frontend/src/Components/StudentMetrics.tsx

```diff
@@ -0,0 +1,197 @@
+                                width={600}
+                                height={300}
+                                data={metrics?.peak_login_times}
+                            >
+                                <CartesianGrid strokeDasharray="3 3" />
+                                <XAxis
+                                    dataKey="time_interval"
+                                    tickFormatter={(tick: string) =>
+                                        new Date(tick).toLocaleTimeString([], {
+                                            hour: '2-digit',
+                                            minute: '2-digit'
+                                        })
+                                    }
+                                />
+                                <YAxis allowDecimals={false} />
+                                <Tooltip />
+                                <Legend />
+                                <Bar dataKey="total_logins" fill="#8884d8" />
+                            </BarChart>
+                        </div>
+                    </div>
+
+                    {/* Filters */}
+                    <div className="flex flex-row gap-6 mb-6">
+                        <div>
+                            <label htmlFor="facility" className="label">
+                                <span className="label-text">Facility</span>
+                            </label>
+                            <select
+                                id="facility"
+                                className="select select-bordered w-full max-w-xs"
+                                value={facility}
+                                onChange={(e) => setFacility(e.target.value)}
+                            >
+                                <option key={'all'} value={'all'}>
+                                    All
+                                </option>
+                                {facilities?.map((facility) => (
+                                    <option
+                                        key={facility.id}
+                                        value={facility.id}
+                                    >
+                                        {facility.name}
+                                    </option>
+                                ))}
+                            </select>
+                        </div>
+                        <div>
+                            <label htmlFor="days" className="label">
+                                <span className="label-text">Days</span>
+                            </label>
+                            <select
+                                id="days"
+                                className="select select-bordered w-full max-w-xs"
+                                value={days}
+                                onChange={(e) =>
+                                    setDays(parseInt(e.target.value))
+                                }
+                            >
```

Comment:
I think visually you could move this up to the same row where the refresh button is- kinda similar to our search/sort and add rows in other pages. so it would look something like this
<img width="1512" alt="Screenshot 2024-12-06 at 12 55 12 PM" src="https://github.com/user-attachments/assets/dcd7a2dd-27ff-4d95-bedd-d6dd84902d04">

Reply:
with the labels too obviously